test/unit/CommandLineController.test.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Tests for the Command-Line Argument Parser
+ */
+
+// When yargs encounters an invalid option, it'll automatically invoke process.exit().
+// This is a problem, because it would muck up our testing framework.
+// We could handle this by explicitly passing a fail() callback to yargs,
+// but this would override the very-nice default handler. In short, testing
+// this scenario would require a lot of code.
+//
+// I don't think that the effort is worth it, because our code is just a
+// wrapper around yargs anyways (I trust their tests).
+
+import * as cli from '../../src/Controller/CommandLineController'
+
+describe('Parser', () => {
+	it('interprets short options', () => {
+		const argv: readonly string[] = [
+			'0',
+			'1',
+			'2',
+			'-f',
+			'mf',
+			'-i',
+			'di',
+			'-t',
+			't',
+		]
+		const results = cli.interpretArgv(argv)
+		expect(results.makefile).toBe('mf')
+		expect(results.docker_image).toBe('di')
+		expect(results.target).toBe('t')
+	})
+
+	it('interprets --option=var style long options', () => {
+		const argv: readonly string[] = [
+			'0',
+			'1',
+			'2',
+			'3',
+			'--makefile=mf',
+			'--docker-image=di',
+			'--target=t',
+		]
+		const results = cli.interpretArgv(argv)
+		expect(results.makefile).toBe('mf')
+		expect(results.docker_image).toBe('di')
+		expect(results.target).toBe('t')
+	})
+
+	it('interprets "--option var" style long options', () => {
+		const argv: readonly string[] = [
+			'0',
+			'1',
+			'2',
+			'3',
+			'--makefile',
+			'mf',
+			'--docker-image',
+			'di',
+			'--target',
+			't',
```

Comment:
Make "mf", "di", and "t" longer so you understand what "type" they are without context, e.g. "build.make", "fake-image:latest", "clean". Also, extract them into variables with names so they're not "magic strings".